File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        loc = []
        char = []
        for i in range(len(s)):
            if s[i] in 'aeiouAEIOU':
                loc.append(i)
                char.append(s[i])
        ans = ''
        cnt = 0
        for i in range(len(s)):
            if cnt < len(loc) and i == loc[cnt]:
                ans += char[-1-cnt]
                cnt += 1
            else:
                ans += s[i]
        return ans

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u'
            || c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
    }

    string reverseVowels(string s) {
        int start = 0;
        int end = s.size() - 1;

        while (start < end) {
            while (start < s.size() && !isVowel(s[start])) {
                start++;
            }
            while (end >= 0 && !isVowel(s[end])) {
                end--;
            }
            if (start < end) {
                swap(s[start++], s[end--]);
            }
        }
        return s; 
    }
};